tests/test_mixins.py

```diff
@@ -46,6 +111,32 @@ def test_create_should_return_different_instances_for_different_classes(
             AnotherSingletonDerived.create() is not SingletonDerived.create()
         )
 
+    def test_create_should_call_get_instance_key(self):
+        with patch.object(
+            SingletonClass, '_get_instance_key'
+        ) as get_instance_key_mock:
+            SingletonClass.create(1, 2, arg3=3)
+            get_instance_key_mock.assert_called_once_with(1, 2, arg3=3)
+
+    def test_create_should_pass_args_to_create_a_new_singleton_instance(self):
+        instance = SingletonClass.create(1, 2, arg3=3)
+
+        assert instance.arg1 == 1
+        assert instance.arg2 == 2
+        assert instance.arg3 == 3
+
+    def test_create_should_return_same_instance_for_equal_args(self):
+        instance_a = SingletonClass.create(1, 2, arg3=3)
+        instance_b = SingletonClass.create(1, 2, arg3=3)
+
+        assert instance_a == instance_b
```

Comment:
To be sure the instance is the same, it is better to use `is` instead of `==` (An instance could implement `__eq__` in a way that would return `True` without being the same instance)

Ex:
```python
assert Decimal(1) == 1  # True
```

```suggestion
        assert instance_a is instance_b
```

Reply:
I fixed the test, thanks!